test/numeric_literal_separators.coffee

```diff
@@ -0,0 +1,63 @@
+# Numeric Literal Separators
+# --------------------------
+
+test 'integer literals with separators', ->
+  eq 123_456, 123456
+  eq 12_34_56, 123456
+
+test 'decimal literals with separators', ->
+  eq 1_2.34_5, 12.345
+  eq 1_0e1_0, 10e10
+  eq 1_2.34_5e6_7, 12.345e67
+
+test 'hexadecimal literals with separators', ->
+  eq 0x1_2_3_4, 0x1234
+
+test 'binary literals with separators', ->
+  eq 0b10_10, 0b1010
+
+test 'octal literals with separators', ->
+  eq 0o7_7_7, 0o777
+
+test 'infinity with separator', ->
+  eq 2e3_08, Infinity
+
+test 'range with separators', ->
+  range = [10_000...10_002]
```

Comment:
I guess I don't have a recent enough version of Node installed to actually run these tests but the compiled JS output was as expected

Reply:
Node 13.5 handles them.